Creating Data Processing GUI initial

hudkinsnoah opened the pull request:
This is the first commit of the Data Processing GUI. There is currently only the right half of the screen displayed which is the two graphs and their dropdowns that they need. Due to there not being anything filled in on the left side for the genetic information (coming soon) everything is spaced closer to center. This will be fixed and respaced once more is added.

hudkinsnoah committed:
Creating Data Processing GUI initial

This is the first commit of the Data Processing GUI. There is currently only the right half of the screen displayed which is the two graphs and their dropdowns that they need. Due to there not being anything filled in on the left side for the genetic information (coming soon) everything is spaced closer to center. This will be fixed and respaced once more is added.

murphan reviewed (approved):
Looks good to me! I like the look of the graph interface and dropdowns.

hudkinsnoah committed:
New Files and GUI/control Logic

Just saving changes up to this point. New files added for the environment GUI and data processing and environment control. Still several changes needed for adding in functionality and making the GUI look cleaner, add in code comments, and set some boundaries to account for differences in screen sizes.

hudkinsnoah committed:
Environment Changes

Adding in temporary changes to the environment for both set up and manipulation. Likely to be some change but this is at least the initial idea for the environment for both looks and manipulation

hudkinsnoah committed:
Secondary idea for Environment

This is a second file that has other ideas for the Environment and how that will be set up and displayed. A lot is commented out just because of it being experimental and playing around with some manipulation of things at the moment.

hudkinsnoah committed:
New ideas and options

This is saving changes to old files that have been updated and adding in likely the new and final option for creating a grid and buttons in the setup_environmentGUI.py file

hudkinsnoah committed:
Saving Changes

Saving changes to repo

hudkinsnoah committed:
Initial GUI creation complete

This is the initial look that the GUI will have. This has the very basics and still has a long ways to go, but it has all pages created, the starting of each of the elements, and all buttons are connected with a function of sorts.

hudkinsnoah committed:
Small changes and additions

This adds in the ability to fill a square when clicked and updating some docstrings

hudkinsnoah committed:
Changes for click type

This will allow the user to select a type of   block that is pressed. It hasn't been hooked to the Environment yet, but it does allow the user to select an option still.

balduvian commented on Control_DataProcessingGUI.py:
```diff
@@ -0,0 +1,101 @@
+    def __init__(self):
+        pass
+
+    def set_graph1_array1(self, data):
```
it seeeeeeems like to me that whatever is going to be implemented here could be done in a smarter way, like having a parameter for graph number and one for array number and only using one function. Unless the data going into each is going to be completely different.

balduvian commented on Control_EnvironmentGUI.py:
```diff
@@ -0,0 +1,132 @@
+        print(f"height: {height}")
+        self.height = height
+
+    def set_temperature(self, temperature):
```
on the c++ side we are moving to using an array of the environmental variables. We'll probably have like 4, but it would probably be smart to have a general setter for any of the environment variables that takes an argument for which variable it sets. You can set up an enum that matches ours eventually, that hasn't been built on the c++ side yet so we don't exactly know which variable names values 0, 1, 2, and 3 will map to yet

balduvian commented on Setup_EnvironmentGUI.py:
```diff
@@ -0,0 +1,97 @@
+        pygame.init()
+        SCREEN = pygame.display.set_mode((WINDOW_WIDTH, WINDOW_HEIGHT))
+        CLOCK = pygame.time.Clock()
+        SCREEN.fill(WHITE)
```
I think it would be better looking if the grid has a black background with very thin lighter gridlines. For now the gridlines give us a good indication of where things are, but it may be a good idea in the future to make a grid toggle. When there's a lot of organisms on screen it's going to look nicer without the gridlines probably.

balduvian commented on Setup_EnvironmentGUI.py:
```diff
@@ -0,0 +1,97 @@
+        # The three buttons that are on the environment screen
+        self.settings_button = Button(SCREEN, (1424, 738), "Settings", (255, 255, 0))
+        self.pause_button = Button(SCREEN, (1364, 738), "Pause", (255, 0, 0))
+        self.play_button = Button(SCREEN, (1320, 738), "Play", (0, 255, 0))
```
the buttons seem to not be wide enough to fit the text on my screen. It could be a scaling problem? They also overlap the grid. To make sure we can add UI elements without obscuring the board, they should probably be on their own bar which sits below the display of the grid.

balduvian reviewed (approved):
Even though there's things that could change, it doesn't affect the rest of the program yet
